src/services/outliningElementsCollector.ts

```diff
@@ -89,6 +106,69 @@ namespace ts.OutliningElementsCollector {
             return isFunctionBlock(node) && node.parent.kind !== SyntaxKind.ArrowFunction;
         }
 
+        function isRegionStart(start: number, end: number) {
+            if (!ts.formatting.getRangeOfEnclosingComment(sourceFile, start, /*onlyMultiLine*/ true)) {
+                const comment = sourceFile.text.substring(start, end);
+                const result = comment.match(regionStart);
+
+                if (result && result.length > 0) {
+                    const sections = result[0].split(" ").filter(function (s) { return s !== ""; });
+
+                    if (sections[0] === "//") {
+                        if (sections.length > 2) {
+                            return result[0].substring(result[0].indexOf(sections[2]));
+                        }
+                        else {
+                            return regionText;
+                        }
+                    }
+                    else {
+                        if (sections.length > 1) {
+                            return result[0].substring(result[0].indexOf(sections[1]));
+                        }
+                        else {
+                            return regionText;
+                        }
+                    }
+                }
+            }
+            return "";
+        }
+
+        function isRegionEnd(start: number, end: number) {

```

Comment:
does this return a `boolean`? Please rename if not, or coerce the result to a boolean via `!!`